src/dune_rules/cram_rules.ml

```diff
@@ -167,6 +169,20 @@ let rules ~sctx ~expander ~dir tests =
                 | None -> acc.alias
                 | Some a -> Alias.Name.Set.add acc.alias a
               in
-              { acc with enabled_if; deps; alias })
+              let packages =
+                match spec.package with
+                | None -> acc.packages
+                | Some (p : Package.t) ->
+                  Package.Name.Set.add acc.packages (Package.Id.name p.id)
+              in
+              { acc with enabled_if; deps; alias; packages })
       in
-      test_rule ~sctx ~expander ~dir effective test)
+      let test_rule () = test_rule ~sctx ~expander ~dir effective test in
+      match !Clflags.only_packages with
+      | None -> test_rule ()
+      | Some only ->
+        if
+          Package.Name.Set.is_empty only
```

Comment:
Note that this test will always be false because we cannot pass an empty set to `--only-packages`:

```
$ dune runtest -f --display quiet --only-packages ""
dune: option `--only-packages': Invalid package name: ""
Usage: dune runtest [OPTION]... [DIR]...
Try `dune runtest --help' or `dune --help' for more information.
```

Since this is dead-code, it seems better to me to not have it.

Reply:
Actually, this was just a type and I meant to check `Package.Name.Set.is_empty effective.pacakges` instead. But thanks for pointing this line of code, I would have otherwise not noticed this bug.

The semantic that we want for tests that aren't attached to a package is that they should never be filtered. This is consistent with how we handle `tests`.

Reply:
Makes sense